aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -184,6 +245,19 @@ private S3Client client() {
     return client;
   }
 
+  private ExecutorService executorService() {
```

Comment:
We should probably shutdown the `ExecutorService` on close of the FileIO

Reply:
Thanks, I have made the changes.

Reply:
I think we cannot close it, because it's a static variable and shared across FileIOs? Which is similar to the one in `S3OutputStream`. We either not close it, or make it an instance variable instead. I think not closing it is fine given we are already doing that in S3OutputStream, any thoughts?

Reply:
I feel not closing is fine. We can keep it as static considering parity with `S3OutputStream`.